src/control-plane/backend/lambda/api/service/reporting.ts

```diff
@@ -117,7 +117,10 @@ export class ReportingServ {
       const dashboardCreateParameters = query.dashboardCreateParameters as DashboardCreateParameters;
 
       //construct parameters to build sql
-      const viewName = query.viewName;
+      let viewName = query.viewName;
+      if (query.action === ExploreRequestAction.PREVIEW) {
```

Comment:
create a shared method for creating temp view name

Reply:
fixed